### PR TITLE
Cache the currently loaded Scenario, instead of re-reading it from the filesystem or (shudder) web. #377

### PR DESCRIPTION
**Pull request by dabreegster**
@michaelkirk, any thoughts on the odd lifetime of `scenario`? I considered carefully cleaning it up, similar to the calls of `app.set_prebaked(None)`, but I think this way is a little simpler.

### CODE REVIEW

**Comment by michaelkirk** [comment]
Not sure if I'm missing something, but this seems straight forward. 

Is your concern that,  since it's now globalish state, you could access `app.primary.scenario` even though it's no longer the "current" scenario?

I suppose we could lock down the API more - like make it private, and have the getter verify the name, or some kind of "get cached else run this block to fetch it" API.

Or you could trust your programmer to not make that mistake. It's only used in this one spot (for now anyway).

**Review by michaelkirk — APPROVED** [review]

**Comment by dabreegster** [comment]
> Is your concern that, since it's now globalish state, you could access app.primary.scenario even though it's no longer the "current" scenario?

Yeah, exactly. I think the current approach is fine for now. But it makes me re-evaluate the approach of having globalish state in `App` at all. Things like `Map` and `DrawMap` being in there make sense; legitimately pretty much every state of the game will use that. But prebaked data and the scenario are only valid under `SandboxMode` and its children. `suspended_sim` is another good example -- it only makes sense once you jump into edit mode. I initially stuck it in `EditMode` and plumbed it to child states, but there wasn't a natural way to pass it between states there. Logically, exactly one state owns it at a time. If each state has it as a field, then when doing `Transition::Push`, you actually have to make a full clone, since `fn event(&mut self` doesn't consume `self`. One hack around that is wrapping it in an `Option` so you can do `.take()`, but then the `State` technically has this weird broken state, where somebody accidentally takes the suspended sim, but doesn't immediately destruct the state.

In other words, we have at least 3 lifetimes for game state:
- the stuff in `App` that should pretty much last the whole program (maybe getting overwritten when a new map is loaded)
- local fields in a particular `State`
- this in-between stuff that's scoped to a few `State`s

I don't think there's anything we should change now, just wanted to get an opinion how maintainable this third category is
